brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -159,9 +160,9 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
                 }
             }
 
-            final HttpClientResponse response = ClientRequestContextAdapter.asHttpClientResponse(log);
+            final HttpClientResponse response = ClientRequestContextAdapter.asHttpClientResponse(log, request);
             try (SafeCloseable ignored = ctx.push()) {
-                handler.handleReceive(response, log.responseCause(), span);
+                handler.handleReceive(response, response.error(), span);
```

Comment:
Unrelated to this PR so just food for thought / another PR but just noticed it - since we pass the span to the handler, so we need to push the context (e.g. mount the scope)?

Reply:
brave 5.11 doesn't mount the scope anymore intentionally, to improve performance (as scope handlers dangling off a thread local can cost a bit). instead the parsers get an arg for the context. the old implementation of the parser will indeed scope, but as handleSend/handleReceive are callbacks (iotw cul-de-sacs, not interceptors which would surround user code), we don't have any use case for scoping anymore internally.

Reply:
Cool - yeah I think that means we can recover the request context push around the callbacks too that is there to allow brave to scope off the threadlocal but if it's not doing that it should be unnecessary.

Reply:
great idea